### PR TITLE
fix(qwik): fixes to better support Qwik SDK

### PR DESCRIPTION
**Pull request by mhevery**
- Don't flatten state getters into state. This allows the state object
  to be removed in most cases, so that Qwik does not need to serialize
  it unnecessarily.
- Support metadata flag to generate light components. This allows us
  to flag which components are light. In our case it will be the
  `<RenderBlock>` components.
- Minor bug-fixes around the way JSX is generated.

## Description

Add a short description of:
- what changes you made, 
- why you made them, and 
- any other context that you think might be helpful for someone to better understand what is contained in this pull request. 

This sort of information is useful for people reviewing the code, as well as anyone from the future trying to understand why changes were made or why a bug started happening.

### CODE REVIEW

File: packages/core/src/generators/qwik/component-generator.ts [hunk 1]
```diff
@@ -6,17 +6,35 @@ import { renderJSXNodes } from './jsx';
...
 import { babelTransformExpression } from '../../helpers/babel-transform';
 import { BaseTranspilerOptions, Transpiler } from '../../types/transpiler';
-import { MitosisNode } from '../../types/mitosis-node';
 
 Error.stackTraceLimit = 9999;
 
 // TODO(misko): styles are not processed.
 
-const DEBUG = false;
+const DEBUG = true;
```
Comment (samijaber) [inline packages/core/src/generators/qwik/component-generator.ts:14]: You forgot to switch this back

File: packages/core/src/generators/qwik/component-generator.ts [hunk 2]
```diff
@@ -317,22 +387,24 @@ function stateToMethodOrGetter(state: Record<string, any>): Record<string, 'meth
...
-    }
-    if (key.startsWith('on') && binding?.code) {
-      const args = binding?.arguments || [];
-      binding.code = `(${args.join(',')}) => ${binding.code}`;
-      delete node.bindings[key];
-      node.bindings[key + '$'] = binding;
+/**
+ * Return a top-level element for the component.
+ *
+ * WHAT: If the component has a single root element, than this returns the element name.
```
Comment (samijaber) [inline packages/core/src/generators/qwik/component-generator.ts:393]:
```suggestion
 * WHAT: If the component has a single root element, then this returns the element name.
```

File: packages/core/src/generators/qwik/component-generator.ts [hunk 3]
```diff
@@ -317,22 +387,24 @@ function stateToMethodOrGetter(state: Record<string, any>): Record<string, 'meth
...
+ *
+ * WHY: This is useful to pull the root element into the component's host and those saving unnecessary wrapping.
```
Comment (samijaber) [inline packages/core/src/generators/qwik/component-generator.ts:395]: Did you mean this instead?

```suggestion
 * WHY: This is useful to pull the root element into the component's host and thus save unnecessary wrapping.
```

**Review by samijaber — APPROVED** [review]

File: packages/core/src/generators/qwik/component-generator.ts (hunk 1 above)
Comment (mhevery) [reply packages/core/src/generators/qwik/component-generator.ts]: fixed

File: packages/core/src/generators/qwik/component-generator.ts (hunk 2 above)
Comment (mhevery) [reply packages/core/src/generators/qwik/component-generator.ts:409]: fixed

File: packages/core/src/generators/qwik/component-generator.ts (hunk 3 above)
Comment (mhevery) [reply packages/core/src/generators/qwik/component-generator.ts:411]: fixed

**Pull request merged**